react/features/google-api/googleApi.js

```diff
@@ -223,6 +225,79 @@ const googleApi = {
             '?part=id%2Csnippet%2Ccdn%2Cstatus',
             `&id=${boundStreamID}`
         ].join('');
+    },
+
+    /**
+     * Retrieves calendar entries from all available calendars.
+     *
+     * @param {number} fetchStartDays - The number of days to go back
+     * when fetching.
+     * @param {number} fetchEndDays - The number of days to fetch.
+     * @returns {Promise<CalendarEntry>}
+     * @private
+     */
+    _getCalendarEntries(fetchStartDays, fetchEndDays) {
+        return this.get()
+            .then(() => this.isSignedIn())
+            .then(isSignedIn => {
+                if (!isSignedIn) {
+                    return null;
+                }
+
+                return this._getGoogleApiClient()
+                    .client.calendar.calendarList.list();
+            })
+            .then(calendarList => {
+                const calendarIds
+                    = calendarList.result.items.map(en => en.id);
+                const promises = calendarIds.map(id => {
+                    const startDate = new Date();
+                    const endDate = new Date();
+
+                    startDate.setDate(startDate.getDate() + fetchStartDays);
+                    endDate.setDate(endDate.getDate() + fetchEndDays);
+
+                    return this._getGoogleApiClient()
+                        .client.calendar.events.list({
```

Comment:
Did you look at the google api request batching? I'm wondering if it's super easy.

Reply:
You mean to get rid of some of the Promises part?

Reply:
No, to make it so there aren't multiple requests going out at once and instead are batched into one request. Now I think we should punt on it considering the breadth of this PR.